comp/dogstatsd/server/server.go

```diff
@@ -816,23 +817,35 @@ func (s *server) parseServiceCheckMessage(parser *parser, message []byte, origin
 	return serviceCheck, nil
 }
 
+// getBuckets converts the buckets from the configuration into slice of float64. The default from the configuration are
+// slice of strings
 func getBuckets(cfg model.Reader, logger log.Component, option string) []float64 {
-	if !cfg.IsSet(option) {
-		return nil
-	}
-
-	buckets, err := cfg.GetFloat64SliceE(option)
```

Comment:
`GetFloat64SliceE` was described as a workaround for a missing feature in viper. Rather than moving it here, can we make it a feature of the new config package? We can change the defaults from `[]string{}` to `[]float64{}` (AIUI the string type was due to viper limitations as well).

Reply:
Yes, this should definitively be part of the config. But right now we're trying to replace viper, once it's done we'll no longer have any restriction and will be able to change default and extend the config API. Until then having a simpler and less "hacky" API help us move forward.

I'll add a comment and we'll come back to it once viper is gone.

Reply:
Great idea :+1:

Reply:
After discussing it internally we reverted to your first proposition about supporting GetFloat64Slice internally. The new version is more flexible than the previous one as more types will be converted to Float64 by the `cast` pacakge.